Orbitersdk/samples/ProjectApollo/src_lm/lm_telecom.cpp

```diff
@@ -2585,35 +2625,40 @@ void LEM_SteerableAnt::Timestep(double simdt){
 
 	//Signal Strength
 
-	VECTOR3 U_RP, pos, R_E, R_M, U_R;
-	MATRIX3 Rot, RX, RY;
-	double relang, Moonrelang;
+	double relang[4], Moonrelang;
+
+	VECTOR3 U_RP[4], pos, R_E, R_M, U_R[4];
+	MATRIX3 Rot;
 
-	//Unit vector of antenna in vessel's local frame
-	RY = _M(cos(pitch), 0.0, sin(pitch), 0.0, 1.0, 0.0, -sin(pitch), 0.0, cos(pitch));
-	RX = _M(1.0, 0.0, 0.0, 0.0, cos(yaw), sin(yaw), 0.0, -sin(yaw), cos(yaw));
-	U_RP = mul(NBSA, mul(RY, mul(RX, _V(0.0, 0.0, 1.0))));
-	U_RP = _V(U_RP.y, U_RP.x, U_RP.z);
+
+	U_RP[0] = LEM_SteerableAnt::pitchYaw2GlobalVector(pitch, yaw - (0.1 * RAD), NBSA);
+	U_RP[1] = LEM_SteerableAnt::pitchYaw2GlobalVector(pitch, yaw + (0.1 * RAD), NBSA);
+	U_RP[2] = LEM_SteerableAnt::pitchYaw2GlobalVector(pitch - (0.1 * RAD), yaw, NBSA);
+	U_RP[3] = LEM_SteerableAnt::pitchYaw2GlobalVector(pitch + (0.1 * RAD), yaw, NBSA);
+	
 
 	//Global position of Earth, Moon and spacecraft, spacecraft rotation matrix from local to global
 	lem->GetGlobalPos(pos);
 	oapiGetGlobalPos(hEarth, &R_E);
 	oapiGetGlobalPos(hMoon, &R_M);
 	lem->GetRotationMatrix(Rot);
 
-	//Calculate antenna pointing vector in global frame
-	U_R = mul(Rot, U_RP);
-	//relative angle between antenna pointing vector and direction of Earth
-	relang = acos(dotp(U_R, unit(R_E - pos)));
+	double EarthSignalDist, LEM_SteerableAntFrequency, LEM_SteerableAntWavelength, Gain85ft, Power85ft;
+	double RecvdLEM_SteerableAntPower, RecvdLEM_SteerableAntPower_dBm;
 
-	if (relang < PI05 / hpbw_factor)
-	{
-		SignalStrength = cos(hpbw_factor*relang)*cos(hpbw_factor*relang)*75.625;
-	}
-	else
-	{
-		SignalStrength = 0.0;
-	}
+	double LEM_SteerableAntGain = pow(16.5 / 10, 10);
+
+	EarthSignalDist = length(pos - R_E) - oapiGetSize(hEarth); //distance from earth's surface in meters
+
+	LEM_SteerableAntFrequency = 2119; //MHz. Should this get set somewhere else?
```

Comment:
Lots of constants here, maybe shouldn't be done on every timestep?

Reply:
yeah, that's probably wasting a bit of time.
Removed.